azurerm/internal/services/storage/resource_arm_hpc_cache_blob_target.go

```diff
@@ -0,0 +1,272 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/storagecache/mgmt/2019-11-01/storagecache"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	storage "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/client"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/parsers"
+	storageValidate "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+	"github.com/tombuildsstuff/giovanni/storage/2018-11-09/blob/containers"
+)
+
+func resourceArmHPCCacheBlobTarget() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmHPCCacheBlobTargetCreateOrUpdate,
+		Update: resourceArmHPCCacheBlobTargetCreateOrUpdate,
+		Read:   resourceArmHPCCacheBlobTargetRead,
+		Delete: resourceArmHPCCacheBlobTargetDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parsers.HPCCacheTargetID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: storageValidate.HPCCacheTargetName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"cache_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"namespace_path": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: storageValidate.HPCCacheNamespacePath,
+			},
+
+			"storage_container_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
since we're able to confirm this is a Resource Manager Resource ID here - can we add more specific validation here?